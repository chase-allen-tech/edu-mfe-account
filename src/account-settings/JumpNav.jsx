import React from 'react';
import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
// import { NavLink } from 'react-router-hash-link';
import { NavLink } from "react-router-dom";
import Scrollspy from 'react-scrollspy';

import { getConfig } from '@edx/frontend-platform';
import PropTypes from 'prop-types';
import messages from './AccountSettingsPage.messages';

function JumpNav({ intl, displayDemographicsLink }) {
  return (
    <div className="jump-nav" style={{paddingTop: 50}}>
      <Scrollspy
        items={[
          '',
          'account',
          // 'demographics-information',
          'preference',
          'document',
          // 'linked-accounts',
          // 'delete-account',
        ]}
        className="list-unstyled"
        currentClassName="font-weight-bold"
      >
        <li className="fsp-12 text-b fw-500 lh-48 ls-1 text-uppercase mb-0">
          <NavLink to="/setting/" className="text-muted text-decoration-none hover-opacity trans-2" activeClassName="text-black">
            {intl.formatMessage(messages['account.settings.section.your.profile'])}
          </NavLink>
        </li>
        <li className="fsp-12 text-b fw-500 lh-48 ls-1 text-uppercase mb-0">
          <NavLink to="/setting/account" className="text-muted text-decoration-none hover-opacity trans-2" activeClassName="text-black">
            {intl.formatMessage(messages['account.settings.section.your.account'])}
          </NavLink>
        </li>
        {/* {getConfig().ENABLE_DEMOGRAPHICS_COLLECTION && displayDemographicsLink
          && (
          <li className="fsp-12 text-b fw-500 lh-48 ls-1 text-uppercase mb-0">
            <NavLink to="/demographics-information" className="text-muted text-decoration-none hover-opacity trans-2" activeClassName="text-black">
              {intl.formatMessage(messages['account.settings.section.demographics.information'])}
            </NavLink>
          </li>
          )} */}
        <li className="fsp-12 text-b fw-500 lh-48 ls-1 text-uppercase mb-0">
          <NavLink to="/setting/preference" className="text-muted text-decoration-none hover-opacity trans-2" activeClassName="text-black">
            {intl.formatMessage(messages['account.settings.section.your.preferences'])}
          </NavLink>
        </li>
        <li className="fsp-12 text-b fw-500 lh-48 ls-1 text-uppercase mb-0">
          <NavLink to="/setting/document" className="text-muted text-decoration-none hover-opacity trans-2" activeClassName="text-black">
            {intl.formatMessage(messages['account.settings.section.your.documents'])}
          </NavLink>
        </li>
        {/* <li className="fsp-12 text-b fw-500 lh-48 ls-1 text-uppercase mb-0">
          <NavLink to="/linked-accounts" className="text-muted text-decoration-none hover-opacity trans-2" activeClassName="text-black">
            {intl.formatMessage(messages['account.settings.section.linked.accounts'])}
          </NavLink>
        </li>
        <li className="fsp-12 text-b fw-500 lh-48 ls-1 text-uppercase mb-0">
          <NavLink to="/delete-account" className="text-muted text-decoration-none hover-opacity trans-2" activeClassName="text-black">
            {intl.formatMessage(messages['account.settings.jump.nav.delete.account'])}
          </NavLink>
        </li> */}
      </Scrollspy>
    </div>
  );
}

JumpNav.propTypes = {
  intl: intlShape.isRequired,
  displayDemographicsLink: PropTypes.bool,
};

export default injectIntl(JumpNav);
