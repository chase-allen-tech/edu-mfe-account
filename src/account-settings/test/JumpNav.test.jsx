import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider, injectIntl } from '@edx/frontend-platform/i18n';

import { BrowserRouter as Router } from 'react-router-dom';
import { mergeConfig, setConfig } from '@edx/frontend-platform';
import JumpNav from '../JumpNav';

const IntlJumpNav = injectIntl(JumpNav);

describe('JumpNav', () => {
  mergeConfig({
    ENABLE_DEMOGRAPHICS_COLLECTION: false,
  });

  let props = {};

  beforeEach(() => {
    props = {
      intl: {},
      displayDemographicsLink: false,
    };
  });

  it('should not render Optional Information link', () => {
    const tree = renderer.create((
      // Had to wrap the following in a router or I will receive an error stating:
      // "Invariant failed: You should not use <NavLink> outside a <Router>"
      <Router>
        <IntlProvider locale="en">
          <IntlJumpNav {...props} />
        </IntlProvider>
      </Router>
    ))
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render Optional Information link', () => {
    setConfig({
      ENABLE_DEMOGRAPHICS_COLLECTION: true,
    });

    props = {
      ...props,
      displayDemographicsLink: true,
    };

    const tree = renderer.create((
      // Same as previous test
      <Router>
        <IntlProvider locale="en">
          <IntlJumpNav {...props} />
        </IntlProvider>
      </Router>
    ))
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
