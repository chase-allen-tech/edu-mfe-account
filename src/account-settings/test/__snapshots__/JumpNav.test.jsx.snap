// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JumpNav should not render Optional Information link 1`] = `
<div
  className="jump-nav"
>
  <ul
    className="list-unstyled"
    style={Object {}}
  >
    <li
      className=""
    >
      <a
        aria-current="page"
        className="active"
        href="/#basic-information"
        onClick={[Function]}
        style={Object {}}
      >
        Account Information
      </a>
    </li>
    <li
      className=""
    >
      <a
        aria-current="page"
        className="active"
        href="/#profile-information"
        onClick={[Function]}
        style={Object {}}
      >
        Profile Information
      </a>
    </li>
    <li
      className=""
    >
      <a
        aria-current="page"
        className="active"
        href="/#social-media"
        onClick={[Function]}
        style={Object {}}
      >
        Social Media Links
      </a>
    </li>
    <li
      className=""
    >
      <a
        aria-current="page"
        className="active"
        href="/#site-preferences"
        onClick={[Function]}
        style={Object {}}
      >
        Site Preferences
      </a>
    </li>
    <li
      className=""
    >
      <a
        aria-current="page"
        className="active"
        href="/#linked-accounts"
        onClick={[Function]}
        style={Object {}}
      >
        Linked Accounts
      </a>
    </li>
    <li
      className=""
    >
      <a
        aria-current="page"
        className="active"
        href="/#delete-account"
        onClick={[Function]}
        style={Object {}}
      >
        Delete My Account
      </a>
    </li>
  </ul>
</div>
`;

exports[`JumpNav should render Optional Information link 1`] = `
<div
  className="jump-nav"
>
  <ul
    className="list-unstyled"
    style={Object {}}
  >
    <li
      className=""
    >
      <a
        aria-current="page"
        className="active"
        href="/#basic-information"
        onClick={[Function]}
        style={Object {}}
      >
        Account Information
      </a>
    </li>
    <li
      className=""
    >
      <a
        aria-current="page"
        className="active"
        href="/#profile-information"
        onClick={[Function]}
        style={Object {}}
      >
        Profile Information
      </a>
    </li>
    <li
      className=""
    >
      <a
        aria-current="page"
        className="active"
        href="/#demographics-information"
        onClick={[Function]}
        style={Object {}}
      >
        Optional Information
      </a>
    </li>
    <li
      className=""
    >
      <a
        aria-current="page"
        className="active"
        href="/#social-media"
        onClick={[Function]}
        style={Object {}}
      >
        Social Media Links
      </a>
    </li>
    <li
      className=""
    >
      <a
        aria-current="page"
        className="active"
        href="/#site-preferences"
        onClick={[Function]}
        style={Object {}}
      >
        Site Preferences
      </a>
    </li>
    <li
      className=""
    >
      <a
        aria-current="page"
        className="active"
        href="/#linked-accounts"
        onClick={[Function]}
        style={Object {}}
      >
        Linked Accounts
      </a>
    </li>
    <li
      className=""
    >
      <a
        aria-current="page"
        className="active"
        href="/#delete-account"
        onClick={[Function]}
        style={Object {}}
      >
        Delete My Account
      </a>
    </li>
  </ul>
</div>
`;
